aws-lc-rs/src/ed25519.rs

```diff
@@ -90,22 +92,56 @@ impl Debug for Ed25519KeyPair {
     }
 }
 
+impl Drop for Ed25519KeyPair {
+    fn drop(&mut self) {
+        self.private_key.zeroize();
+    }
+}
+
 #[derive(Clone)]
 #[allow(clippy::module_name_repetitions)]
-pub struct PublicKey {
-    public_key: [u8; ED25519_PUBLIC_KEY_LEN],
+/// The seed value for the `EdDSA` signature scheme using Curve25519
+pub struct Seed<'a>(&'a Ed25519KeyPair);
+
+/// Elliptic curve private key data encoded as a big-endian fixed-length integer.
+#[allow(clippy::module_name_repetitions)]
+pub struct Ed25519SeedBuffer {
+    _priv: (),
+}
+
+impl Seed<'_> {
+    /// Exposes the seed encoded as a big-endian fixed-length integer.
+    ///
+    /// For most use-cases, `EcdsaKeyPair::to_pkcs8()` should be preferred.
+    ///
+    /// # Errors
+    /// `error::Unspecified` if serialization failed.
+    pub fn to_buffer(&self) -> Result<Buffer<'static, Ed25519SeedBuffer>, Unspecified> {
+        let buffer = Vec::from(&self.0.private_key[..ED25519_PRIVATE_KEY_SEED_LEN]);
+        Ok(Buffer::<Ed25519SeedBuffer>::new(buffer))
+    }
+}
+
+impl Debug for Seed<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str("Ed25519Seed()")
+    }
 }
 
+#[derive(Clone)]
+#[allow(clippy::module_name_repetitions)]
+pub struct PublicKey([u8; ED25519_PUBLIC_KEY_LEN]);
```

Comment:
Should we expose this as a Buffer instead? I think before this PR the PublicKey type wasn't directly public, only visible in the public API via the KeyPair trait, so we have that flexibility if we want it.

But no strong feeling either way.